x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/CloseJobRequest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.protocol.xpack.ml;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.unit.TimeValue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class CloseJobRequest extends ActionRequest {
+
+    public static final String ALL_JOBS = "_all";
+
+    private List<String> jobIds = new ArrayList<>();
+    private TimeValue timeout;
+    private boolean force;
+    private boolean allowNoJobs = true;
+
+    public static CloseJobRequest closeAllJobsRequest(){
+        return new CloseJobRequest(ALL_JOBS);
+    }
+
+    public CloseJobRequest(String jobId) {
+        jobIds.add(Objects.requireNonNull(jobId, "[jobId] must not be null"));
+    }
+
+    CloseJobRequest() {
+    }
+
+    public void setJobIds(List<String> jobIds) {
+        Objects.requireNonNull(jobIds, "jobIds must not be null");
+        if (jobIds.stream().anyMatch(Objects::isNull)) {
+            throw new NullPointerException("[jobId] must not be null");
+        }
+        this.jobIds = new ArrayList<>(jobIds);
+    }
+
+    public void addJobId(String jobId) {
+        jobIds.add(Objects.requireNonNull(jobId, "[jobId] must not be null"));
+    }
+
+    public List<String> getJobIds() {
+        return jobIds;
+    }
+
+    public TimeValue getTimeout() {
+        return timeout;
+    }
+
+    public void setTimeout(TimeValue timeout) {
+        this.timeout = timeout;
+    }
+
+    public boolean isForce() {
+        return force;
+    }
+
+    public void setForce(boolean force) {
+        this.force = force;
+    }
+
+    public boolean isAllowNoJobs() {
+        return allowNoJobs;
+    }
+
+    public void setAllowNoJobs(boolean allowNoJobs) {
+        this.allowNoJobs = allowNoJobs;
+    }
+
+    public String getCommaDelimitedJobIdString() {
```

Comment:
I wonder if this really belongs in the request converter. It is the request logic that requires this bit of processing so we could simply call this utility method in the request conversion.